### PR TITLE
Math magicians: Events Review

### PR DESCRIPTION
**Pull request by Felipeg005**
To achieve project requirements next actions were taken:

- [x] Copy the files calculate.js and operate.js into a logic/ directory.
- [x] Analyze the files calculate.js and operate.js.
- [x] Import the files in your Calculator component.
- [x] Implement the event handlers needed to use the math logic from calculate.js and operate.js in the React component.
- [x] Use state and props, pass props to child components, use lifecycle methods, lift the state up.
- [x] Test the app in the browser. It's able to perform math operations.

[Link to live version](https://felipeg005.github.io/Math-Magicians-React-App/)

😄 Thanks for your review.

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,30 +1,52 @@
...
+    await this.setState(calculate(this.state, e.target.innerHTML));
+    if (this.state.total !== null && this.state.next === null) {
+      document.querySelector('.display-board').innerHTML = this.state.total;
+    } else if (this.state.next !== null && this.state.total !== null) {
+      document.querySelector('.display-board').innerHTML = this.state.next;
+    } else if (this.state.next !== null && this.state.total === null) {
+      document.querySelector('.display-board').innerHTML = this.state.next;
+    } else if (this.state.next === null && this.state.total === null) {
+      document.querySelector('.display-board').innerHTML = '0';
+    }
```
Comment (george-shammar) [inline src/components/calculator.js:24]: - [ ] Your application looks great already. When I clicked any operate button twice or more times, I received this error : `Unhandled Rejection (Error): [big.js] Invalid number` . Please, let's create an exception to handle this scenario.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @Felipeg005 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js (hunk 1 above)
Comment (Felipeg005) [reply src/components/calculator.js]: Thanks for finding that BUG 🐛 , now I create an exception in the code to fix it. 😄

**Review by oxenprogrammer — APPROVED** [review]
Hi **@Felipeg005**, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

**Optional Changes**
- [x] Your README has broken images and links. You want to ensure that it is always up to date since it acts as your living documentation.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**